### PR TITLE
CP-1044 enable public CI, w_module

### PR DESCRIPTION
**Pull request by trentgrover-wf**
## Issue
- Enable travisCI + Codecov.io
- update pubspec.yaml
- add badges to README

## Changes
**Source:**
- none

**Tests:**
- none

## Areas of Regression
- none

## Testing
- none

## Code Review
@maxwellpeterson-wf
@evanweible-wf
@dustinlessard-wf 
@jayudey-wf

### CODE REVIEW

**Comment by trentgrover-wf** [comment]
@jayudey-wf once this is merged, please push this to pub

**Comment by maxwellpeterson-wf** [comment]
+1

File: .gitignore [hunk 1]
```diff
@@ -5,9 +5,10 @@
 .pub
 build
 packages
-.packages
 pubspec.lock
 
-# coverage
-/coverage/
+# docs
+dartdoc-viewer/
```
Comment (evanweible-wf) [inline .gitignore]: This is now `/doc/api/` with the new dartdoc tool.

Comment (trentgrover-wf) [inline .gitignore]: fixed

**Comment by dustinlessard-wf** [comment]
+1

**Comment by maxwellpeterson-wf** [comment]
+1

**Comment by evanweible-wf** [comment]
+1

**Comment by trentgrover-wf** [comment]
@jayudey-wf ready for merge and publish to pub

**Comment by jayudey-wf** [comment]
QA Resource Approval: +10

- [ ] Testing instruction
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
 - verified passing travis build
- [ ] Unit test created/updated
- [ ] All unit tests pass

Merging into master.

**Pull request merged**